### DISCUSSION TITLE
Python Uno Card Game

### DISCUSSION THREAD

**Discussion by vbarbar** [community/community#68167] [Codespaces]
### Select Topic Area

Question

### Body

I am making an UNO card game and I generated and then shuffled a deck for it and each player is then given five cards from the shuffled deck. All the cards are supposed to be string objects in the lists but the wild card is being stored as a list object instead of a string object. I want it to be stored as a string.

import random

def buildDeck():
    deck = []
    colours = ["Red", "Green", "Yellow", "Blue"]
    values = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, "Draw Two", "Skip", "Reverse"]
    wilds = ["Wild", "Wild Draw Four"]
    
    for colour in colours:
        for value in values:
            cardVal = "{} {}".format(colour, value)
            deck.append(cardVal)
            if value != 0:
                deck.append(cardVal)
    
    for i in range(4):
        deck.append([wilds[0]])
        deck.append(wilds[1])
    
    return deck

def shuffleDeck(deck):
    for cardPos in range(len(deck)):
        randPos = random.randint(0, 107)
        deck[cardPos], deck[randPos] = deck[randPos], deck[cardPos]
    return deck

def drawCards(numCards):
    cardsDrawn = []
    for x in range (numCards):
        cardsDrawn.append(unoDeck.pop(0))
    return cardsDrawn

def showHand(player, playerHand):
    print("{}'s turn".format(player))
    print("Your Hand")
    print("-----------------------------")
    y = 1
    for card in playerHand:
        print("{}) {}".format(y, card))
        y += 1
    print("")

unoDeck = buildDeck()
unoDeck = shuffleDeck(unoDeck)
discards = []

players = [] #List to store player cards
playerNames = [] #List to store player names
colours = ["Blue", "Red", "Green", "Yellow"]
numPlayers = None
print("Enter --help to display the rules of the game\n")

numPlayers = input("How many players? ")
if numPlayers == "--help" or numPlayers == "--resume":
    checkInput(numPlayers)
else:
    numPlayers = int(numPlayers)
    while len(playerNames) < numPlayers:
        tempName = input("Enter player's name: ")
        if tempName == "--help" or tempName == "--resume":
            checkInput(tempName)
        else:
            playerNames.append(tempName)
            players.append(drawCards(5))

print("The cards are:")
for (x,y) in zip(playerNames, players):
    print("Player {} has {}".format(x, y))
print("")

**Comment by talentdev219** [comment]
for i in range(4):
        deck.append([wilds[0]]) # <-- has extra brackets
        deck.append(wilds[1])

but this can also be simplified to

# build the rest of the deck

for i in range(4):
    deck += wilds
or

# build the rest of the deck

deck += wild * 4
also the random module has a built in shuffle method: https://docs.python.org/3/library/random.html#random.shuffle which will do an inplace shuffle. so you can do:

uno_deck = buildDeck()
random.shuffle(uno_deck)

**Reply by vbarbar** [reply]
Thank you.
This is a correct answer that I want.

**Comment by Akash1134** [comment]
GitHub's [Acceptable Use Policies](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies) prohibits coordinated or inauthentic activity like rapid questions and answers. As a result, we'll be unmarking the answer and locking this post. Please note any future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.
